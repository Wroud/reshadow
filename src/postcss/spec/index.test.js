const postcss = require('postcss');
const {stripIndent} = require('common-tags');

const reshadow = require('..');

const createTransform = (options = {}) => {
    const processor = postcss([reshadow(options)]);

    return css => processor.process(stripIndent(css), {from: undefined}).css;
};

const transform = createTransform();

describe('postcss', () => {
    it('should transform the code', () => {
        const code = transform`
            button {
                color: red;
            }

            button[disabled] {
                color: grey;
            }

            button[type="submit"] {
                border: 1px solid red;
            }
        `;

        expect(code).toMatchSnapshot();
    });

    it('should transform the code with namespaces', () => {
        const code = transform`
            |button,
            use|button {
                color: red;
            }

            button use|content,
            button |content {
                font-weight: bold;
            }

            button[|disabled],
            button[use|disabled] {
                color: grey;
            }

            button[|size="m"],
            button[use|size="m"] {
                font-size: 14px;
            }
        `;

        expect(code).toMatchSnapshot();
    });

    it('should should keep elements under html namespace', () => {
        const code = transform`
            content html|button {
                color: red;
            }
        `;

        expect(code).toMatchSnapshot();
    });

    it('should should not transform keyframes', () => {
        const code = transform`
            @keyframes anim {
                from {
                    transform: translateX(-10px);
                }
                to {
                    transform: translateX(0);
                }
            }
        `;

        expect(code).toMatchSnapshot();
    });

    describe('stats', () => {
        const transform = createTransform({stats: true});

        it('should transform the code with stats', () => {
            const code = transform`
                [type="submit"] {
                    padding: 10px;
                }

                button {
                    color: red;
                }

                button[disabled] {
                    color: grey;
                }

                button[type="submit"] {
                    border: 1px solid red;
                }

                button[use|size="s"] {
                    font-size: 14px;
                }
            `;

            expect(code).toMatchSnapshot();
        });
    });

    describe('bem', () => {
        const transform = createTransform({bem: true});

        it.skip('should transform the code with bem', () => {
            const code = transform`
                __button {
                    color: red;
                }

                __button[disabled] {
                    color: grey;
                }

                __button[_size="m"] {
                    font-size: 14px;
                }
            `;

            expect(code).toMatchSnapshot();
        });
    });
});
