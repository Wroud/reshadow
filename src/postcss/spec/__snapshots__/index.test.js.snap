// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss should should keep elements under the :global 1`] = `
".__content button {
    color: red;
}

.__button[aria-hidden] {
    opacity: 0;
}

.__button[disabled]._use--variant._use--variant_normal {
    background-color: gray;
}"
`;

exports[`postcss should should not transform keyframes 1`] = `
"@keyframes anim {
    from {
        transform: translateX(-10px);
    }
    to {
        transform: translateX(0);
    }
}"
`;

exports[`postcss should should not transform pseudo-classes like :nth-child 1`] = `
".__tr:nth-child(2n) {
    color: red;
}

:dir(rtl) {
    background-color: red;
}"
`;

exports[`postcss should transform the code 1`] = `
".__button {
    color: red;
}

.__button._disabled {
    color: grey;
}

.__button._type._type_submit {
    border: 1px solid red;
}"
`;

exports[`postcss should transform the code with namespaces 1`] = `
".__use--button,.__use--button {
    color: red;
}

.__button .__use--content,.__button .__use--content {
    font-weight: bold;
}

.__button._use--disabled,.__button._use--disabled {
    color: grey;
}

.__button._use--size._use--size_m,.__button._use--size._use--size_m {
    font-size: 14px;
}"
`;

exports[`postcss stats should transform the code with stats 1`] = `
"._type._type_submit {
    padding: 10px;
}

.__button {
    color: red;
}

.__button._disabled {
    color: grey;
}

.__button._type._type_submit {
    border: 1px solid red;
}

.__button._use--size._use--size_s {
    font-size: 14px;
}

@value __elements__: '{\\"__common__\\":{\\"mods\\":{},\\"props\\":{\\"type\\":[\\"submit\\"]}},\\"button\\":{\\"mods\\":{\\"use--size\\":[\\"s\\"]},\\"props\\":{\\"disabled\\":[\\"\\"],\\"type\\":[\\"submit\\"]}}}'"
`;
